### DISCUSSION TITLE
Confused about relationships

### DISCUSSION THREAD

**Discussion by chrber04** [prisma/prisma#12846] [Q&A] [answered]
hi guys, new to Prisma and me and my friend are having some doubts if what we are doing is correct. I'll let the code explain itself:

```
model User {
  id String @id @default(cuid())

  created_referral_codes ReferralCode[] @relation("created_referral_codes") // The referral codes created by this user.

  used_referral_code    ReferralCode? @relation(fields: [used_referral_code_id], references: [id], name: "referrals")
  used_referral_code_id String?
}
```

```
model ReferralCode {
  id String @id @default(cuid())

  creator_id String @unique // The creator of this referral code.
  creator    User   @relation(fields: [creator_id], references: [id], name: "created_referral_codes")

  referrals      User[] @relation(name: "referrals") // The referrals that has joined through this referral code.
}
```

Very certain about the "created_referral_codes" one to many relationship on the user model, not as certain about the "referrals" relationship on the referralcode however. prisma is secretly creating them, but the problem is, those arent really accessible it seems?

`ALTER TABLE "users" ADD CONSTRAINT "users_referral_fkey" FOREIGN KEY ("referral") REFERENCES "referrals_for_referral_code"("id") ON DELETE SET NULL ON UPDATE CASCADE;`

**Comment by austincrim** [comment, marked as answer]
Hey @chrber04 👋 ,

I think something like this makes the most sense:
```prisma
model User {
  id String @id @default(cuid())

  created_referral_codes ReferralCode[] @relation("created_referral_codes")

  used_referral_code    ReferralCode @relation(fields: [used_referral_code_id], references: [used_by_user_id])
  used_referral_code_id String       @unique
}

model ReferralCode {
  id                 String  @id @default(cuid())
  created_by_user    User?   @relation(fields: [created_by_user_id], references: [id], name: "created_referral_codes")
  created_by_user_id String?

  used_by_user    User?
  used_by_user_id String? @unique
}
```

Let me know if that makes more sense.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
